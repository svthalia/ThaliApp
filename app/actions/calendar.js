export const OPEN = 'CALENDAR_OPEN';
export const REFRESH = 'CALENDAR_REFRESH';
export const SUCCESS = 'CALENDAR_SUCCESS';
export const FAILURE = 'CALENDAR_FAILURE';
export const FETCHING = 'CALENDAR_FETCHING';

export function open() {
  return {
    type: OPEN,
  };
}

export function refresh(keywords) {
  return {
    type: REFRESH,
    payload: { keywords },
  };
}

export function success(eventList) {
  return {
    type: SUCCESS,
    payload: { eventList },
  };
}

export function failure() {
  return { type: FAILURE };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}
