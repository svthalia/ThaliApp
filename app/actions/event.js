export const EVENT = 'EVENT_EVENT';
export const OPEN = 'EVENT_OPEN';
export const FETCHING = 'EVENT_FETCHING';
export const SUCCESS = 'EVENT_SUCCESS';
export const DONE = 'EVENT_DONE';
export const FAILURE = 'EVENT_FAILURE';
export const ADMIN = 'EVENT_ADMIN';
export const UPDATE_REGISTRATION = 'EVENT_UPDATE_REGISTRATION';
export const ADD_TO_CALENDAR = 'EVENT_ADD_CALENDAR';

export function event(pk, navigateToEventScreen = true) {
  return {
    type: EVENT,
    payload: { pk, navigateToEventScreen },
  };
}

export function open() {
  return {
    type: OPEN,
  };
}

export function success(eventData, eventRegistrations) {
  return {
    type: SUCCESS,
    payload: { eventData, eventRegistrations },
  };
}

export function done() {
  return {
    type: DONE,
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}

export function admin() {
  return {
    type: ADMIN,
  };
}

export function updateRegistration(pk, present, payment) {
  return {
    type: UPDATE_REGISTRATION,
    payload: { pk, present, payment },
  };
}

export function addToCalendar(eventName, eventLocation, eventStart, eventEnd) {
  return {
    type: ADD_TO_CALENDAR,
    payload: {
      eventName,
      eventLocation,
      eventStart,
      eventEnd,
    },
  };
}
