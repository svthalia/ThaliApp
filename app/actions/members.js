export const OPEN = 'MEMBERS_OPEN';
export const MEMBERS = 'MEMBERS_MEMBERS';
export const FETCHING = 'MEMBERS_FETCHING';
export const MEMBERS_SUCCESS = 'MEMBERS_MEMBERS_SUCCESS';
export const FAILURE = 'MEMBERS_FAILURE';
export const MORE = 'MEMBERS_MORE';
export const MORE_SUCCESS = 'MEMBERS_MORE_SUCCESS';

export function open() {
  return {
    type: OPEN,
  };
}

export function members(keywords = '') {
  return {
    type: MEMBERS,
    payload: { keywords },
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function success(memberList, next, searchKey) {
  return {
    type: MEMBERS_SUCCESS,
    payload: { memberList, next, searchKey },
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}

export function more(url) {
  return {
    type: MORE,
    payload: { url },
  };
}

export function moreSuccess(memberList, next) {
  return {
    type: MORE_SUCCESS,
    payload: { memberList, next },
  };
}
