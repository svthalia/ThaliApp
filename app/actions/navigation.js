export const BACK = 'NAVIGATE_BACK';
export const TOGGLE_DRAWER = 'NAVIGATE_TOGGLE_DRAWER';
export const OPEN_WEBSITE = 'NAVIGATE_OPEN_WEBSITE';

export function goBack() {
  return {
    type: BACK,
  };
}

export function toggleDrawer() {
  return {
    type: TOGGLE_DRAWER,
  };
}

export function openWebsite(url) {
  return {
    type: OPEN_WEBSITE,
    payload: url,
  };
}
