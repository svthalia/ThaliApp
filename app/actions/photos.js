export const PHOTOS_ALBUMS_OPEN = 'PHOTOS_ALBUMS_OPEN';
export const PHOTOS_ALBUMS_LOAD = 'PHOTOS_ALBUMS_LOAD';
export const PHOTOS_ALBUMS_SUCCESS = 'PHOTOS_ALBUMS_SUCCESS';
export const PHOTOS_ALBUMS_FAILURE = 'PHOTOS_ALBUMS_FAILURE';
export const PHOTOS_ALBUMS_FETCHING = 'PHOTOS_ALBUMS_FETCHING';
export const PHOTOS_ALBUM_OPEN = 'PHOTOS_ALBUM_OPEN';
export const PHOTOS_ALBUM_SUCCESS = 'PHOTOS_ALBUM_SUCCESS';
export const PHOTOS_ALBUM_FAILURE = 'PHOTOS_ALBUM_FAILURE';
export const PHOTOS_ALBUM_FETCHING = 'PHOTOS_ALBUM_FETCHING';
export const PHOTOS_GALLERY_OPEN = 'PHOTOS_GALLERY_OPEN';

export function openAlbums() {
  return {
    type: PHOTOS_ALBUMS_OPEN,
  };
}

export function loadAlbums(keywords, next) {
  return {
    type: PHOTOS_ALBUMS_LOAD,
    payload: { keywords, next },
  };
}

export function successAlbums(data, next, isNext) {
  return {
    type: PHOTOS_ALBUMS_SUCCESS,
    payload: { data, next, isNext },
  };
}

export function failureAlbums() {
  return { type: PHOTOS_ALBUMS_FAILURE };
}

export function fetchingAlbums() {
  return {
    type: PHOTOS_ALBUMS_FETCHING,
  };
}

export function openGallery(selection) {
  return {
    type: PHOTOS_GALLERY_OPEN,
    payload: { selection },
  };
}

export function openAlbum(pk) {
  return {
    type: PHOTOS_ALBUM_OPEN,
    payload: { pk },
  };
}

export function openAlbumWithSlug(slug) {
  return {
    type: PHOTOS_ALBUM_OPEN,
    payload: { slug },
  };
}

export function successAlbum(data) {
  return {
    type: PHOTOS_ALBUM_SUCCESS,
    payload: { data },
  };
}

export function failureAlbum() {
  return { type: PHOTOS_ALBUM_FAILURE };
}

export function fetchingAlbum() {
  return {
    type: PHOTOS_ALBUM_FETCHING,
  };
}
