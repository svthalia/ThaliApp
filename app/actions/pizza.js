export const PIZZA = 'PIZZA_PIZZALIST';
export const FETCHING = 'PIZZA_FETCHING';
export const SUCCESS = 'PIZZA_SUCCESS';
export const FAILURE = 'PIZZA_FAILURE';
export const CANCEL = 'PIZZA_CANCEL';
export const CANCEL_SUCCESS = 'PIZZA_CANCEL_SUCCESS';
export const ORDER = 'PIZZA_ORDER';
export const ORDER_SUCCESS = 'PIZZA_ORDER_SUCCESS';
export const ADMIN = 'PIZZA_ADMIN';
export const ADMIN_ORDERS = 'PIZZA_ADMIN_ORDERS';
export const ADMIN_UPDATE_ORDER = 'PIZZA_ADMIN_UPDATE_ORDER';
export const ADMIN_LOADING = 'PIZZA_ADMIN_LOADING';
export const ADMIN_SUCCESS = 'PIZZA_ADMIN_SUCCESS';
export const ADMIN_FAILURE = 'PIZZA_ADMIN_FAILURE';

export function retrievePizzaInfo() {
  return {
    type: PIZZA,
  };
}

export function success(event, order, pizzaList) {
  return {
    type: SUCCESS,
    payload: { event, order, pizzaList },
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}

export function cancelOrder() {
  return {
    type: CANCEL,
  };
}

export function cancelSuccess() {
  return {
    type: CANCEL_SUCCESS,
  };
}

export function orderPizza(pk, hasOrder) {
  return {
    type: ORDER,
    payload: { pk, hasOrder },
  };
}

export function orderSuccess(order) {
  return {
    type: ORDER_SUCCESS,
    payload: { order },
  };
}

export function openAdmin() {
  return {
    type: ADMIN,
  };
}

export function retrieveOrders() {
  return {
    type: ADMIN_ORDERS,
  };
}

export function updateOrder(pk, payment) {
  return {
    type: ADMIN_UPDATE_ORDER,
    payload: { pk, payment },
  };
}

export function adminLoading() {
  return {
    type: ADMIN_LOADING,
  };
}

export function adminSuccess(orders) {
  return {
    type: ADMIN_SUCCESS,
    payload: { orders },
  };
}

export function adminFailure() {
  return {
    type: ADMIN_FAILURE,
  };
}
