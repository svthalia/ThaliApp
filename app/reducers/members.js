import * as memberActions from '../actions/members';

const initialState = {
  memberList: [],
  status: 'initial',
  loading: false,
  more: null,
  searchKey: '',
};

export default function loadEvent(state = initialState, action = {}) {
  switch (action.type) {
    case memberActions.FETCHING: {
      return {
        ...state,
        loading: true,
      };
    }
    case memberActions.MEMBERS_SUCCESS: {
      return {
        ...state,
        status: 'success',
        loading: false,
        memberList: action.payload.memberList,
        more: action.payload.next,
        searchKey: action.payload.searchKey,
      };
    }
    case memberActions.FAILURE: {
      return {
        ...state,
        status: 'failure',
        loading: false,
      };
    }
    case memberActions.MORE_SUCCESS: {
      return {
        ...state,
        status: 'success',
        loading: false,
        memberList: [...state.memberList, ...action.payload.memberList],
        more: action.payload.next,
      };
    }
    default:
      return state;
  }
}
