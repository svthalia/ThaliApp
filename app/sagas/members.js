import { Dimensions } from 'react-native';
import { call, put, select, takeEvery } from 'redux-saga/effects';

import { TOTAL_BAR_HEIGHT } from '../ui/components/standardHeader/style/StandardHeader';
import { memberSize } from '../ui/screens/memberList/style/MemberList';

import { apiRequest } from '../utils/url';
import * as memberActions from '../actions/members';
import { tokenSelector } from '../selectors/session';
import reportError from '../utils/errorReporting';

const members = function* members(action) {
  const { keywords } = action.payload;
  const token = yield select(tokenSelector);

  yield put(memberActions.fetching());

  const { height } = Dimensions.get('window');
  const amountOfRows = Math.floor((height - TOTAL_BAR_HEIGHT) / (memberSize + 16));

  const data = {
    method: 'GET',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
  };

  const params = {
    limit: amountOfRows * 6,
  };
  if (keywords) {
    params.search = keywords;
  }

  try {
    const response = yield call(apiRequest, 'members', data, params);
    yield put(memberActions.success(response.results, response.next, keywords));
  } catch (error) {
    yield call(reportError, error);
    yield put(memberActions.failure());
  }
};

const more = function* more(action) {
  const { url } = action.payload;
  const token = yield select(tokenSelector);

  yield put(memberActions.fetching());

  const data = {
    method: 'GET',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
  };

  try {
    const response = yield call(apiRequest, url, data);
    yield put(memberActions.moreSuccess(response.results, response.next));
  } catch (error) {
    yield call(reportError, error);
    yield put(memberActions.moreSuccess([], null));
  }
};

export default function* membersSaga() {
  yield takeEvery(memberActions.MEMBERS, members);
  yield takeEvery(memberActions.MORE, more);
}
