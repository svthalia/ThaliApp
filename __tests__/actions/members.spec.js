import * as actions from '../../app/actions/members';

describe('member actions', () => {
  it('should expose the member actions', () => {
    expect(actions.OPEN).toEqual('MEMBERS_OPEN');
    expect(actions.MEMBERS).toEqual('MEMBERS_MEMBERS');
    expect(actions.FETCHING).toEqual('MEMBERS_FETCHING');
    expect(actions.MEMBERS_SUCCESS).toEqual('MEMBERS_MEMBERS_SUCCESS');
    expect(actions.FAILURE).toEqual('MEMBERS_FAILURE');
    expect(actions.MORE).toEqual('MEMBERS_MORE');
    expect(actions.MORE_SUCCESS).toEqual('MEMBERS_MORE_SUCCESS');
  });

  it('should create an action to open the member list', () => {
    expect(actions.members()).toMatchSnapshot();
  });

  it('should create an action to notify the member list should be fetched unfiltered', () => {
    expect(actions.members()).toMatchSnapshot();
  });

  it('should create an action to notify the member list should be fetched, filtered by certain keywords', () => {
    expect(actions.members('John Doe')).toMatchSnapshot();
  });

  it('should create an action to notify the member list is being fetched', () => {
    expect(actions.fetching()).toMatchSnapshot();
  });

  it('should create an action to notify the member list was fetched successfully', () => {
    expect(
      actions.success(['John Doe', 'Jane Doe'], 'nextUrl', 'John Doe')
    ).toMatchSnapshot();
  });

  it('should create an action to notify of a failure fetching the member list', () => {
    expect(actions.failure()).toMatchSnapshot();
  });

  it('should create an action to notify additional members should be retrieved', () => {
    expect(actions.more('nextUrl')).toMatchSnapshot();
  });

  it('should create an action to notify the additional members were fetched successfully', () => {
    expect(actions.moreSuccess(['John Doe', 'Jane Doe'], 'nextUrl')).toMatchSnapshot();
  });
});
