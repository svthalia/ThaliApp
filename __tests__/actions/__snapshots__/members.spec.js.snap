// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`member actions should create an action to notify additional members should be retrieved 1`] = `
Object {
  "payload": Object {
    "url": "nextUrl",
  },
  "type": "MEMBERS_MORE",
}
`;

exports[`member actions should create an action to notify of a failure fetching the member list 1`] = `
Object {
  "type": "MEMBERS_FAILURE",
}
`;

exports[`member actions should create an action to notify the additional members were fetched successfully 1`] = `
Object {
  "payload": Object {
    "memberList": Array [
      "John Doe",
      "Jane Doe",
    ],
    "next": "nextUrl",
  },
  "type": "MEMBERS_MORE_SUCCESS",
}
`;

exports[`member actions should create an action to notify the member list is being fetched 1`] = `
Object {
  "type": "MEMBERS_FETCHING",
}
`;

exports[`member actions should create an action to notify the member list should be fetched unfiltered 1`] = `
Object {
  "payload": Object {
    "keywords": "",
  },
  "type": "MEMBERS_MEMBERS",
}
`;

exports[`member actions should create an action to notify the member list should be fetched, filtered by certain keywords 1`] = `
Object {
  "payload": Object {
    "keywords": "John Doe",
  },
  "type": "MEMBERS_MEMBERS",
}
`;

exports[`member actions should create an action to notify the member list was fetched successfully 1`] = `
Object {
  "payload": Object {
    "memberList": Array [
      "John Doe",
      "Jane Doe",
    ],
    "next": "nextUrl",
    "searchKey": "John Doe",
  },
  "type": "MEMBERS_MEMBERS_SUCCESS",
}
`;

exports[`member actions should create an action to open the member list 1`] = `
Object {
  "payload": Object {
    "keywords": "",
  },
  "type": "MEMBERS_MEMBERS",
}
`;
